src/components/ChromeRoute/ChromeRoute.tsx

```diff
@@ -41,7 +44,7 @@ const ChromeRoute = memo(
            */
           dispatch(updateDocumentTitle(defaultTitle || 'Hybrid Cloud Console'));
         }
-        dispatch(changeActiveModule(scope));
+        setActiveModule(scope);
```

Comment:
@Hyperkid123 I think Jotai works great for this use case - simple global state management. However in my research I'm finding that folks with more complex use cases still needed something such as Zustand for more operations/actions/state management. See https://docs.pmnd.rs/zustand/getting-started/comparison#jotai for a good comparison. A good example would be an incrementer - where we want to control how the state is updated. I think it's possible in Jotai - it would just require more custom shared logic.

That said - are we suggesting Jotai replace _all_ of redux? Or just part of it?

Reply:
We want to replace everything redux related.

Looking at the comparison I see outdated bits, like with the external store (which Jotai has and is in this PR).

The overall goal is not to obfuscate the actual state mutation and what is happening for example in a reducer. Or a store. Or a selector. The data should be prepared in the component (or some generic functions) and the atom should be updated with the final value.

The fact that jotai is so "dumb" forces us to be smart.

> I think it's possible in Jotai - it would just require more custom shared logic that would require enforcement on our end.

Exactly the goal. We write our own standarts and enforcements instead of relying on something provided. One nice example of this is the selector magic. It claims to optimize rendering but it never mentions that the second one uses `{...}` and does not include often very expensive compare functions the entire optimization goes out of window. And your component will be re-rendered **on each store update if do not have shallow objects with only primitive types**.

For example
```jsx
const state = {
  chrome: {
    nav: {
      isOpen: false
    },
    schemas: {
      foo: {},
      bar: {}
    },
    activeModule: 'foo'
  }
}

const Component = () => {
  const navSchema = useSelector((state) => {
  // creating new object, and creating shallow copy of a state object totally kills the shallow equal what selector based managers are doing
  // plus it hurts to read this
  // selector will trigger render on any state mutation regardless if the mutated state is actually used
    return {
      ...state.chrome.schemas[state.chrome.activeModule],
      isOpen: state.chrome.nav.isOpen
    }
  }, /** We would to implement some strange comparator to  optimize renders*/)
  return (
   ...
  )
}
```

Using atom based approach does not give us the option to create new state objects. It forces us to write the code how we should with no shortcuts. With the selectors we are just kicking the can down the road hoping that at some point someone will optimize the selectors.

Yes, it will be more work, but I believe ultimately it will pay dividents in both performance and code readability.